packages/auth/src/providers/cognito/utils/oauth/getRedirectUrl.native.ts

```
@@ -4,14 +4,27 @@
 import { invalidRedirectException } from '../../../../errors/constants';
 
 /** @internal */
-export function getRedirectUrl(redirects: string[]): string {
-	const redirectUrl = redirects?.find(
+export function getRedirectUrl(
+	redirects: string[],
+	preferredSignOutUrl?: string,
+): string {
+	let preferredRedirectUrl;
+	// Always check for a non http/s url (appScheme)
+	const appSchemeRedirectUrl = redirects?.find(
 		redirect =>
 			!redirect.startsWith('http://') && !redirect.startsWith('https://'),
 	);
-	if (!redirectUrl) {
+	if (!appSchemeRedirectUrl) {
 		throw invalidRedirectException;
 	}
+	if (preferredSignOutUrl) {
+		preferredRedirectUrl = redirects?.find(
+			redirect => redirect === preferredSignOutUrl,
+		);
+		if (!preferredRedirectUrl) {
+			throw invalidRedirectException;
+		}
+	}
```

Couldn't this have used the exact same strategy as the web version?
```ts
if (preferredSignOutUrl) {
	const redirectUrl = redirects?.find(
		redirect => redirect === _preferredSignOutUrl,
	);

	if (!redirectUrl) {
		throw invalidRedirectException;
	}

	return redirectUrl;
}

// existing previous logic
```

---

Yes it can be the same, but in the mobile case we should fail early if there is no app scheme found in the array. It does matter what the preferred url is if there is no app scheme in the array for us to register. In the web version, the preferred url takes precedence.